library/models/product.py

```diff
@@ -0,0 +1,361 @@
+            "unique(barcode,code)",
+            "Barcode and Code must be unique across\
+                          all the products!",
+        )
+    ]
+
+    @api.onchange("is_ebook", "attach_ebook")
+    def onchange_availablilty(self):
+        """Onchange method to define book availability"""
+        if self.is_ebook and self.attach_ebook:
+            self.availability = "available"
+
+    def action_purchase_order(self):
+        """Method to redirect at book order"""
+        purchase_line_obj = self.env["purchase.order.line"]
+        purchase = purchase_line_obj.search([("product_id", "=", self.id)])
+        action = self.env.ref("purchase.purchase_form_action")
+        result = action.read()[0]
+        if not purchase:
+            raise ValidationError(_("There is no Books Purchase !"))
+        order = []
+        [order.append(order_rec.order_id.id) for order_rec in purchase]
+        if len(order) != 1:
+            result["domain"] = "[('id', 'in', " + str(order) + ")]"
+        else:
+            res = self.env.ref("purchase.purchase_order_form", False)
+            result["views"] = [(res and res.id or False, "form")]
+            result["res_id"] = purchase.order_id.id
+        return result
+
+    def action_book_req(self):
+        """Method to request book"""
+        for rec in self:
+            book_req = self.env["library.book.request"].search(
+                [("name", "=", rec.id)]
+            )
+            action = self.env.ref("library.action_lib_book_req")
+            result = action.read()[0]
+            if not book_req:
+                raise ValidationError(_("There is no Book requested"))
+            req = []
+            [req.append(request_rec.id) for request_rec in book_req]
+            if len(req) != 1:
+                result["domain"] = "[('id', 'in', " + str(req) + ")]"
+            else:
+                res = self.env.ref("library.view_book_library_req_form", False)
+                result["views"] = [(res and res.id or False, "form")]
+                result["res_id"] = book_req.id
+            return result
+
+
+class BookAttachment(models.Model):
+    """Defining Book Attachment."""
+
+    _name = "book.attachment"
+    _description = "Stores attachments of the book"
+
+    name = fields.Char("Description", required=True, help="Enter Description")
+    product_id = fields.Many2one(
+        "product.product", "Product", help="Select Book"
+    )
+    date = fields.Date(
+        "Attachment Date", required=True, default=fields.Datetime.today()
+    )
+    attachment = fields.Binary("Attachment", help="Attach attachment here")
+
+
+class LibraryAuthor(models.Model):
+    _inherit = "library.author"
+
+    book_ids = fields.Many2many(
+        "product.product",
+        "author_book_rel",
+        "author_id",
+        "product_id",
+        "Books",
+        help="Related books",
+    )
+
+
+class BookEditor(models.Model):
+    """Book Editor Information"""
+
+    _name = "book.editor"
```

Comment:
can't we use res.partner here?
 cc @AnuKPatel

Reply:
yes, We can. @Trivedi-Vacha-SerpentCS 
@ForamSerpentCS Please check.